sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -242,6 +244,12 @@ def wait_for_idle(self, seconds_to_wait=None):
                 self.logger.error(error)
                 raise LivyClientTimeoutException(error)
 
+            if constants.YARN_RESOURCE_LIMIT_MSG in self._logs and \
```

Comment:
Searching the logs for a particular string in English seems like a bad way to go in the long-term.  This functionality breaks if at any point in the future the error message gets changed on the Livy side even by 1 character.  Is there any other (more structured API-driven) way to accomplish the same goal?